src/CoreWCF.NetTcp/src/CoreWCF/Channels/Framing/FramingConnection.cs

```diff
@@ -22,6 +23,7 @@ public class FramingConnection
     {
         private readonly ConnectionContext _context;
         private IDuplexPipe _transport;
+        private Lazy<IPEndPoint> _remoteEndpoint;
```

Comment:
Use of Lazy isn't needed as we always generate a RemoteEndpointMessageProperty for every request on NetTcp so it's inevitable this will always be used. The only benefit of a Lazy is to make the cost happen at first use rather than connection. Although this has given me an idea for a new feature. What do you think about adding a feature where if you add a `Lazy<T>` message property, then getting the property causes MessageProperties to evaluate the Lazy and return the value. This way things like RemoteEndpointMessageProperty would only get constructed if application code used it.

Reply:
The lazy evaluation of properties might be beneficial if there can go something wrong during creation. The benefit I see there, is that if this really can happen the exception would be properly handed over to the caller instead of happening somewhere within the CoreWCF infrastructure during context creation. 

I remember the case in some old ASP.net Web API times where access to some network level properties could lead to an exception if the client closed the connection already. 

Unfortunately I fear that I'm lacking of the insight into the possible MessageProperties within WCF to judge whether such a lazy evaluation would be worth the effort. Might be a micro optimization without much benefit. For real critical paths, I would assume the lazy evaluation happens within the C# properties of the related MessageProperty class.